### PR TITLE
Commands: Add a brand spanking new `commit-stats` command

### PR DESCRIPTION
**Pull request by filiphsps**
Introducing `/commit-stats`
image

Oh wow, this command can be yours for just as little as $0, yes, the offer is even valid in Nebraska!
Now ain't that a damn good deal!!

### CODE REVIEW

**Comment by filiphsps** [comment]
@networkException - tagging per request

**Review by networkException — CHANGES_REQUESTED** [review]
This is a really neat command!

Sadly it doesn't seem to be very reliable with counting commits.. (at least I'm pretty sure that I have commits in `serenity` :^) )

image

Maybe (ab)using GitHub's pagination api with 1 item per page + getting the number of pages (see https://docs.github.com/en/rest/overview/resources-in-the-rest-api#link-header and perhaps https://www.npmjs.com/package/parse-link-header) would be more robust.

**Comment by filiphsps** [comment]
> This is a really neat command!
> 
> Sadly it doesn't seem to be very reliable with counting commits.. (at least I'm pretty sure that I have commits in `serenity` :^) )
> 
> image
> 
> Maybe (ab)using GitHub's pagination api with 1 item per page + getting the number of pages (see [docs.github.com/en/rest/overview/resources-in-the-rest-api#link-header](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#link-header) and perhaps [npmjs.com/package/parse-link-header](https://www.npmjs.com/package/parse-link-header)) would be more robust.

Alright, this has finally been resolved. I've also made the command a lot more reliable by allowing `getCommitsCount` multiple attempts while also respecting the github ratelimiting timeout.

I had initially planned to add a quick and dirty in-memory cache but decided that against it since it's highly unlikely somebody will run the command with the same parameters in a short time span. The `fetchSerenityRepos` function could be cached in the future, however.

(The rate limiting is solvable by instead using github's webhooks and our own database, might be interesting in the future but way out of scope for this PR). 🤠

File: src/apis/githubAPI.ts [hunk 1]
```diff
@@ -99,8 +109,105 @@ class GithubAPI {
...
+                                author,
+                                repository,
+                                ++attempt
+                            );
+                            resolve(results);
+                        } catch (e) {
+                            reject(e);
+                        }
+                    }, timeout);
+                });
```
Comment (networkException) [inline src/apis/githubAPI.ts:188]: can we do something like

```suggestion
                await new Promise(resolve => setTimeout(resolve, timeout));
                
                // try { ...
```

instead?

**Review by networkException — APPROVED** [review]
One minor nit, LGTM otherwise

File: src/apis/githubAPI.ts (hunk 1 above)
Comment (filiphsps) [reply src/apis/githubAPI.ts]: Done 🤠

**Review by networkException — APPROVED** [review]

**Pull request merged**